pkg/driver/mount.go

```diff
@@ -33,16 +33,18 @@ import (
 )
 
 const (
-	keyIdEnv           = "AWS_ACCESS_KEY_ID"
-	accessKeyEnv       = "AWS_SECRET_ACCESS_KEY"
-	regionEnv          = "AWS_REGION"
-	defaultRegionEnv   = "AWS_DEFAULT_REGION"
-	stsEndpointsEnv    = "AWS_STS_REGIONAL_ENDPOINTS"
-	MountS3PathEnv     = "MOUNT_S3_PATH"
-	defaultMountS3Path = "/usr/bin/mount-s3"
-	procMounts         = "/host/proc/mounts"
-	userAgentPrefix    = "--user-agent-prefix"
-	csiDriverPrefix    = "s3-csi-driver/"
+	keyIdEnv             = "AWS_ACCESS_KEY_ID"
+	accessKeyEnv         = "AWS_SECRET_ACCESS_KEY"
+	regionEnv            = "AWS_REGION"
+	defaultRegionEnv     = "AWS_DEFAULT_REGION"
+	stsEndpointsEnv      = "AWS_STS_REGIONAL_ENDPOINTS"
+	MountS3PathEnv       = "MOUNT_S3_PATH"
+	awsMaxAttemptsEnv    = "AWS_MAX_ATTEMPTS"
+	defaultMountS3Path   = "/usr/bin/mount-s3"
+	procMounts           = "/host/proc/mounts"
+	userAgentPrefix      = "--user-agent-prefix"
+	awsMaxAttemptsOption = "--aws-max-attempts"
```

Comment:
Why are we adding the max attempts configuration as an option as well as an environment variable? What makes it different from something like `AWS_REGION` or `AWS_ACCESS_KEY_ID`?

Reply:
We need max attempts to be configurable by the user on a per PersistentVolume basis, so we have to accept it similar to other mount options. Unlike the other mount options, it's not just passed through to mountpoint and we have to do a conversion to remove the option from the CLI flags and pass it as an environment variable instead.

`AWS_REGION` we get from the driver's environment and that's passed through unchanged to mountpoint. `AWS_ACCESS_KEY_ID` is also unique where we get the key from a k8s secret and pass that through to the driver. Overall there's just different ways that the CSI driver gets options with different ways mountpoint accepts them and the driver has to glue everything together.